docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,209 @@
+Z01_MNAME_NOT_AUTHORITATIVE   | WARNING | ns                   | SOA MNAME name server "{ns}" is not authoritative for the zone.
+Z01_MNAME_NOT_IN_NS_LIST      | INFO    | nsname               | SOA MNAME name server "{nsname}" is not listed as NS record for the zone.
+Z01_MNAME_NOT_MASTER          | WARNING | ns, ns_ip_list       | SOA MNAME name server "{ns}" does not have the highest SERIAL. Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_NOT_RESOLVE         | WARNING | nsname               | SOA MNAME name server "{nsname}" cannot be resolved into an IP address.
+Z01_MNAME_UNEXPECTED_RCODE    | WARNING | ns, rcode            | SOA MNAME name server "{ns}" gives unexpected RCODE name ("{rcode}") in response to an SOA query.
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in [DNS Query and Response Defaults].
+The handling of the DNS responses on the DNS queries follow, unless otherwise specified below,
+what is specified for [DNS Response] in the same specification.
+
+1. Create a [DNS Query] with query type SOA and query name *Child Zone* ("SOA Query").
+
+2. Create the following empty sets:
+   1. Name server IP address, name, SERIAL ("MNAME Nameservers")
+   2. Name server IP address, SERIAL ("SERIAL Nameservers")
+   3. Name server IP address ("MNAME Not Master")
+   4. Name server IP address ("MNAME Is Master")
+   5. Name server IP address ("MNAME Is Localhost")
+   6. Name server IP address ("MNAME Is Dot")
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5] ("Name Server IP").
+
+4. For each name server IP address in *Name Server IP* do:
+   1. Send *SOA Query* to the name server and collect the response.
+   2. Go to next name server IP address if at least one of the following criteria is met:
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no SOA record with owner name matching the query.
+   3. From the DNS response, extract the name server name from the MNAME field:
+      1. If the name is "localhost", then add name server IP address to the *MNAME Is Localhost* set and go to next name server.
+      2. Else if the name is ".", then add name server IP address to the *MNAME Is Dot* set and go to next name server.
+      3. Else, add name server name to the *MNAME Nameservers* set.
+   4. From the DNS response, add the SERIAL field and name server IP address to the *SERIAL Nameservers* set.
+   5. Go to next name server.
```

Comment:
Yes. I thought it would not be useful to do the SERIAL check when the MNAME refers to an unexpected name server name. Plus step 8 cannot be applied to such name server names. If you think otherwise I would be interested to know why.

Reply:
The SOA SERIAL that you collect in 4.4 is not the SOA SERIAL that is found in the SOA MNAME server, but the SOA SERIAL found in the servers listed in **Name Server IP**. I do not see why that should be affected by the SOA MNAME value.

Reply:
True, thanks, I got mixed up after the last update. I will make changes accordingly.